app/src/main/java/bankingapp/LogIn.java

```diff
@@ -101,13 +53,13 @@ public void checkAccount(Connection con, String accountNumber, String pinNumber)
 
                 case 2:
                     System.out.println("Enter income:");
-                    income = scanner.nextInt();
+                    income = input.nextInt();
                     DBTrans.addIncome(con, income, accountNumber);
                     System.out.println("Income was added!");
                     break;
 
```

Comment:
Notice this pattern you have here for case 2.

1. Gather input from the user
2. Basic validation (required fields, required format, etc) (None used in this case.)
2. Feed the input into the domain layer
3. Extract a result and display it to the user.

This is exactly what UI does! Anything not included in that list of four is not UI code, and belongs somewhere separate. This allows you to modify your UI and domain independently.